test/prefer-export-from.mjs

```diff
@@ -289,3 +289,108 @@ test.typescript({
 	],
 	invalid: [],
 });
+
+// `ignoreUsedVariables`
+test.snapshot({
+	valid: [
+		outdent`
+			import defaultExport from 'foo';
+			use(defaultExport);
+			export default defaultExport;
+		`,
+		outdent`
+			import defaultExport from 'foo';
+			use(defaultExport);
+			export {defaultExport};
+		`,
+		outdent`
+			import {named} from 'foo';
+			use(named);
+			export {named};
+		`,
+		outdent`
+			import {named} from 'foo';
+			use(named);
+			export default named;
+		`,
+		outdent`
+			import * as namespace from 'foo';
+			use(namespace);
+			export {namespace};
+		`,
+		outdent`
+			import * as namespace from 'foo';
+			use(namespace);
+			export default namespace;
+		`,
+		outdent`
+			import * as namespace from 'foo';
+			export {namespace as default};
+			export {namespace as named};
+		`,
+		outdent`
+			import * as namespace from 'foo';
+			export default namespace;
+			export {namespace as named};
+		`,
+		outdent`
+			import defaultExport, {named} from 'foo';
+			use(defaultExport);
+			export {named};
+		`,
+		outdent`
+			import defaultExport, {named} from 'foo';
+			use(named);
+			export {defaultExport};
+		`,
+		outdent`
+			import {named1, named2} from 'foo';
+			use(named1);
+			export {named2};
+		`,
+		outdent`
+			import defaultExport, {named1, named2} from 'foo';
+			use(defaultExport);
+			export {named1, named2};
+		`,
+		outdent`
+			import defaultExport, {named1, named2} from 'foo';
+			use(named1);
+			export {defaultExport, named2};
+		`,
+		outdent`
+			import {notUsedNotExported, exported} from 'foo';
+			export {exported};
+		`,
```

Comment:
In this case,  `notUsedNotExported` is not really used, but also not exported,

I think the main purpose for this option is to avoid multiple `from ...`, if variable is not used, it's possible this file not done yet. 

Once user fixes [`no-unused-vars`](https://eslint.org/docs/rules/no-unused-vars), we'll know it will be used or exported.

Reply:
Can you add what you commented here as a comment to that test case? For posterity.

Reply:
On second thought, we can just skip auto-fix, seems better.

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1590/commits/a58be750ceffd0528db2140d45a5e563f6b8bb13